File: abes_pps/data structure/binary_search.cpp
#include <iostream>
using namespace std;
int bin(int a[],int m,int low, int high,int search_key){
    int mid{};
while(low<=high){
    mid=(low+high)/2;
    if(a[mid]==search_key){
        return mid;
    }
    else if(search_key<a[mid]){
        high=mid-1;
    }
    else{
        low=mid+1;
    }
}
return -1;
}
int main(){
int m{6};
int search_key{7};
int a[]={3,5,7,8,9,10};
int low{0};
int high{5};
int mid{};
cout<<bin(a, m,low,high,search_key);
    return 0;
}

File: ABES-EC/sem4/python_lab/binary_search.py
def binary_search(arr, x):
    low = 0
    high = len(arr) - 1
    mid = 0
    while low <= high:
        mid = (high + low) // 2
        if arr[mid] < x:
            low = mid + 1
        elif arr[mid] > x:
            high = mid - 1
        else:
            return mid
    return -1
arr = list(map(int, input("Enter multiple values: ").split()))  
x=int(input("Enter the number to find"))
result = binary_search(arr, x)
if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")
